backend/common/src/main/java/ai/verta/modeldb/common/db/migration/Migrator.java

```diff
@@ -22,19 +31,147 @@ public Migrator(Connection connection, String resourcesDirectory) {
   /** Assumes that the underlying database has already been created. */
   public void performMigration(RdbConfig config) throws SQLException, MigrationException {
     log.info("Starting database migration process");
-    MigrationDatastore migrationDatastore = MigrationDatastore.create(config, connection);
+    performMigration(config, null);
+  }
 
+  void performMigration(RdbConfig config, Integer desiredVersion)
+      throws SQLException, MigrationException {
+    MigrationDatastore migrationDatastore = setupDatastore(config);
+    MigrationState currentState = findCurrentState();
+
+    if (currentState.isDirty()) {
```

Comment:
it looks like `findCurrentState()` can return null, so should there be a null check here?

Reply:
If we have a null current state, then something has failed during initialization. The null check would just throw an IllegalStateException, so not sure that's preferable to the NPE that we'd get now?  I'm not against it, though, if you think that would be clearer.

Reply:
I think that's fine. 👍

Reply:
I'd rather see an explicit IllegalStateException with a good message, rather than having to decrypt an NPE from the logs. Will add!